integreat_cms/cms/utils/zammad.py

```diff
@@ -0,0 +1,243 @@
+            message["content"] = message["body"]
+            reduced_message = {
+                key: message[key] for key in message if key in keys_to_keep
+            }
+            messages.append(reduced_message)
+        return messages
+
+    def save_message(
+        self, message: str, internal: bool, automatic_message: bool
+    ) -> bool:
+        """
+        Add a new message
+
+        :param message: message text to be saved
+        :param internal: true if message shoud not be visible to app user
+        :param automatic_message: true if message does not originate from a human
+        :return: success
+        """
+        cache.delete(f"{self.region.slug}_{self.device_id}")
+        return (
+            self.zammad_request(
+                "POST",
+                "/api/v1/ticket_articles",
+                {
+                    "ticket_id": self.zammad_id,
+                    "body": message,
+                    "internal": internal,
+                    "automatic_message": automatic_message,
+                    "subject": (
+                        "app user message"
+                        if not automatic_message
+                        else "automatically generated message"
+                    ),
+                    "content_type": "text/html",
+                    "type": "web",
+                    "sender": "Customer" if not automatic_message else "Agent",
+                },
+            ).status_code
+            == 200
+        )
+
+    @property
+    def evaluation_consent(self) -> bool:
+        """
+        Get user evaluation consent
+
+        :return: user evaluation consent
+        """
+        return cache.get(
+            f"chat_evaluation_consent_{self.device_id}",
+            self.zammad_request(
+                "GET",
+                f"/api/v1/tickets/{self.zammad_id}",
+            ).json()["evaluation_consent"],
+        )
+
+    def save_evaluation_consent(self, value: bool) -> bool:
+        """
+        Set user evaluation consent
+
+        :param value: True if user agrees, false if not
+        :return: success
+        """
+        cache.delete(f"chat_evaluation_consent_{self.device_id}")
+        return (
+            self.zammad_request(
+                "PUT",
+                f"/api/v1/tickets/{self.zammad_id}",
+                {"evaluation_consent": value},
+            ).status_code
+            == 200
+        )
+
+    @property
+    def automatic_answers(self) -> bool:
+        """
+        Check if automatic answers are turned on/off
+
+        :return: generate automatic answers or not
+        """
+        return cache.get(
+            f"chat_automatic_anwers_{self.device_id}",
+            self.zammad_request(
+                "GET",
+                f"/api/v1/tickets/{self.zammad_id}",
+            ).json()["automatic_answers"],
+        )
+
+    def save_automatic_answers(self, value: bool) -> bool:
```

Comment:
Now that we're using @property instead of @cached_property, it should be possible to use the @evaluation_consent.setter.